spinnaker/spinnaker_system/google_front50_test.py

```diff
@@ -0,0 +1,235 @@
+    config = self.agent.deployed_config
+    enabled = config.get('spinnaker.gcs.enabled', False)
+    if not enabled:
+      raise ValueError('spinnaker.gcs.enabled is not True')
+
+    bucket = config['spinnaker.gcs.bucket']
+    root = config['spinnaker.gcs.rootFolder']
+    self.BUCKET_ROOT = os.path.join(bucket, root)
+    self.TEST_APP = self.bindings['TEST_APP']
+    self.gcs_observer = gcp.GsutilAgent();
+
+    result = self.gcs_observer.run(['versioning', 'get', 'gs://' + bucket])
+    if result.exit_code:
+      raise RuntimeError('Could not inspect bucket: ' + str(result))
+    self.versioning_enabled = result.output.endswith('Enabled')
+    if not self.versioning_enabled:
+      self.logger.info('bucket=%s versioning enabled=%s',
+                       bucket, self.versioning_enabled);
+
+  def __init__(self, bindings, agent=None):
+    """Constructor.
+
+    Args:
+      bindings: [dict] The data bindings to use to configure the scenario.
+      agent: [Front50Agent] The agent for invoking the test operations on
+          Front50
+    """
+    self.logger = logging.getLogger(__name__)
+    super(GoogleFront50TestScenario, self).__init__(bindings, agent)
+
+    self.initial_app_spec = {
+      "name" : self.TEST_APP,
+      "description" : "My Application Description.",
+      "email" : "test@google.com",
+      "accounts" : "my-aws-account,my-google-account",
+      "updateTs" : "1463667655844",
+      "createTs" : "1463666817476",
+      "platformHealthOnly" : False,
+      "cloudProviders" : "gce,aws"        
+    }
+
+
+  def create_app(self):
+    payload = self.agent.make_json_payload_from_object(self.initial_app_spec)
+    expect = dict(self.initial_app_spec)
+    expect['name'] = self.initial_app_spec['name'].upper()
+
+    contract = jc.Contract()
+
+    # Note that curiosly the updated timestamp is not adjusted in the storage
+    # file.
+    gcs_builder = gcp.GoogleCloudStorageContractBuilder(self.gcs_observer)
+    (gcs_builder.new_clause_builder('Created Google Cloud Storage File')
+     .list_path(os.path.join(self.BUCKET_ROOT, 'applications'))
+     .contains_path_value('name', self.TEST_APP))
+    (gcs_builder.new_clause_builder('Wrote File Content')
+     .inspect_contents(
+          os.path.join(self.BUCKET_ROOT, 'applications', self.TEST_APP,
+                       'specification.json'))
+     # 'application-metadata.json'))
+     .contains_path_eq('', expect))
+    for clause in gcs_builder.build().clauses:
+      contract.add_clause(clause)
+
+    # The update timestamp is determined by the server,
+    # and we dont know what that is, so lets ignore it
+    # and assume the unit tests verify it is properly updated.
+    expect = dict(expect)
+    del(expect['updateTs'])
+    f50_builder = st.http_observer.HttpContractBuilder(self.agent)
+    (f50_builder.new_clause_builder('Added Application')
+     .get_url_path(os.path.join('/default/applications/name',
```

Comment:
Added a comment. This is the front50 web server URL to retrieve the application. It isnt directly GCS. I thought I might as well test the web server GET while I was here.